docs/htmlapps.md

```diff
@@ -54,34 +54,93 @@ It's important to "cut" the source websites content into appropriately sized chu
 
 
 
-HTML Writer utility class
--------------------------
+The `HTMLWriter` utility class
+------------------------------
 The class `HTMLWriter` in `ricecooker.utils.html_writer` provides a basic helper
-methods for creating files within a zip file.
+methods for creating files within a zip file to be used as `HTML5Zip` files
+added to `HTML5App` nodes.
+
+To use the `HTMLWriter` class, you must enter the `HTMLWriter` context:
+
+```python
+from ricecooker.utils.html_writer import HTMLWriter
+with HTMLWriter('./myzipfile.zip') as zipper:
+    # Add your code here
+```
+
+To write the main file (`index.html` in the root of the zip file), use the 
+`write_index_contents` method:
+
+```python
+contents = "<html><head></head><body>Hello, World!</body></html>"
+zipper.write_index_contents(contents)
+```
+
+You can also add other files (images, stylesheets, etc.) using `write_file`,
+`write_contents`, and `write_url` methods:
+
+```python
+# Returns path to file "styles/style.css"
+css_path = zipper.write_contents("style.css", "body{padding:30px}", directory="styles")
+extra_head = "<link href='{}' rel='stylesheet'></link>".format(css_path)         # Can be inserted into <head>
+
+img_path = zipper.write_file("path/to/img.png")                                  # Note: file must be local
+img_tag = "<img src='{}'>...".format(img_path)                                   # Can be inserted as image
+
+script_path = zipper.write_url("src.js", "http://example.com/src.js", directory="src")
+script = "<script src='{}' type='text/javascript'></script>".format(script_path) # Can be inserted into html
+```
+
+To check if a file exists in the zipfile, use the `contains` method:
+
+```python
+# Zipfile has "index.html" file
+zipper.contains('index.html')     # Returns True
+zipper.contains('css/style.css')  # Returns False
+```
+
+See the source code for more details:
```

Comment:
^ ^ ^ this was some useful docs from the `souschef.md` which I saved here so as not to throw away